votingpool/input_selection_wb_test.go

```diff
@@ -0,0 +1,403 @@
+	c2 := TstCreateInputs(t, store2, pkScript, []int64{int64(dustThreshold)})[0]
+	// the calculation of if it has been confirmed does this:
+	// chainheigt - bh + 1 >= target, which is quite weird, but the
+	// reason why I need to put 902 as *that* makes 1000 - 902 +1 = 99 >=
+	// 100 false
+	c2.BlockHeight = int32(902)
+	if pool.isCreditEligible(c2, eligibleInputMinConfirmations, chainHeight, dustThreshold) {
+		t.Errorf("Input is eligible and it should not be.")
+	}
+
+}
+
+func TestCreditSortingByAddress(t *testing.T) {
+	teardown, _, pool := TstCreatePool(t)
+	defer teardown()
+
+	series := []TstSeriesDef{
+		{ReqSigs: 2, PubKeys: TstPubKeys[1:4], SeriesID: 0},
+		{ReqSigs: 2, PubKeys: TstPubKeys[3:6], SeriesID: 1},
+	}
+	TstCreateSeries(t, pool, series)
+
+	c0 := TstNewFakeCredit(t, pool, 0, 0, 0, []byte{0x00, 0x00}, 0)
+	c1 := TstNewFakeCredit(t, pool, 0, 0, 0, []byte{0x00, 0x00}, 1)
+	c2 := TstNewFakeCredit(t, pool, 0, 0, 0, []byte{0x00, 0x01}, 0)
+	c3 := TstNewFakeCredit(t, pool, 0, 0, 0, []byte{0x01, 0x00}, 0)
+	c4 := TstNewFakeCredit(t, pool, 0, 0, 1, []byte{0x00, 0x00}, 0)
+	c5 := TstNewFakeCredit(t, pool, 0, 1, 0, []byte{0x00, 0x00}, 0)
+	c6 := TstNewFakeCredit(t, pool, 1, 0, 0, []byte{0x00, 0x00}, 0)
+
+	randomCredits := [][]Credit{
+		[]Credit{c6, c5, c4, c3, c2, c1, c0},
+		[]Credit{c2, c1, c0, c6, c5, c4, c3},
+		[]Credit{c6, c4, c5, c2, c3, c0, c1},
+	}
+
+	want := []Credit{c0, c1, c2, c3, c4, c5, c6}
+
+	for _, random := range randomCredits {
+		sort.Sort(byAddress(random))
+		got := random
+
+		if len(got) != len(want) {
+			t.Fatalf("Sorted credit slice size wrong: Got: %d, want: %d",
+				len(got), len(want))
+		}
+
+		for idx := 0; idx < len(want); idx++ {
+			if !reflect.DeepEqual(got[idx], want[idx]) {
+				t.Errorf("Wrong output index. Got: %v, want: %v",
+					got[idx], want[idx])
+			}
+		}
+	}
+}
+
+// TstFakeCredit is a structure implementing the Credit interface used to test
+// the byAddress sorting. It exists because to test the sorting properly we need
+// to be able to set the Credit's TxSha and OutputIndex.
+type TstFakeCredit struct {
+	addr        WithdrawalAddress
+	txSha       *wire.ShaHash
+	outputIndex uint32
+	amount      btcutil.Amount
+}
+
+func (c *TstFakeCredit) String() string             { return "" }
+func (c *TstFakeCredit) TxSha() *wire.ShaHash       { return c.txSha }
+func (c *TstFakeCredit) OutputIndex() uint32        { return c.outputIndex }
+func (c *TstFakeCredit) Address() WithdrawalAddress { return c.addr }
+func (c *TstFakeCredit) Amount() btcutil.Amount     { return c.amount }
+func (c *TstFakeCredit) TxOut() *wire.TxOut         { return nil }
+func (c *TstFakeCredit) OutPoint() *wire.OutPoint {
+	return &wire.OutPoint{Hash: *c.txSha, Index: c.outputIndex}
+}
+
+func TstNewFakeCredit(t *testing.T, pool *Pool, series uint32, index Index, branch Branch, txSha []byte, outputIdx int) *TstFakeCredit {
+	var hash wire.ShaHash
+	copy(hash[:], txSha)
```

Comment:
Probably not really a big deal since it's unlikely to change anytime soon, but this is assuming the structure of the ShaHash type.  The more correct way to do this would be to call `hash.SetBytes(txsha)` which can throw an error if the format of the bytes were to ever change.